fe/fe-core/src/main/java/com/starrocks/catalog/system/sys/SysFeLocks.java

```diff
@@ -0,0 +1,153 @@
+                        .column("lock_type", ScalarType.createVarcharType(64))
+                        .column("lock_object", ScalarType.createVarcharType(64))
+                        .column("lock_mode", ScalarType.createVarcharType(64))
+                        .column("lock_start_time", ScalarType.createVarcharType(64))
+                        .column("thread_info", ScalarType.createVarcharType(64))
+                        .column("granted", ScalarType.createVarchar(64))
+                        .column("waiter_list", ScalarType.createVarcharType(SystemTable.NAME_CHAR_LEN))
+                        .build(),
+                TSchemaTableType.SYS_FE_LOCKS);
+    }
+
+    @VisibleForTesting
+    public static TFeLocksRes listLocks(TFeLocksReq request, boolean authenticate) throws TException {
+        TAuthInfo auth = request.getAuth_info();
+        UserIdentity currentUser;
+        if (auth.isSetCurrent_user_ident()) {
+            currentUser = UserIdentity.fromThrift(auth.getCurrent_user_ident());
+        } else {
+            currentUser = UserIdentity.createAnalyzedUserIdentWithIp(auth.getUser(), auth.getUser_ip());
+        }
+
+        // authorize
+        try {
+            if (authenticate) {
+                Authorizer.checkSystemAction(currentUser, null, PrivilegeType.OPERATE);
+            }
+        } catch (AccessDeniedException e) {
+            throw new TException(e.getMessage(), e);
+        }
+
+        TFeLocksRes response = new TFeLocksRes();
+        Collection<Database> dbs = GlobalStateMgr.getCurrentState().getFullNameToDb().values();
+        for (Database db : CollectionUtils.emptyIfNull(dbs)) {
+            TFeLocksItem item = resolveLockInfo(db);
+            response.addToItems(item);
+        }
+        return response;
+    }
+
+    @VisibleForTesting
+    protected static TFeLocksItem resolveLockInfo(Database db) {
+        var lock = db.getLock();
+        TFeLocksItem lockItem = new TFeLocksItem();
+        lockItem.setLock_type("DATABASE");
+        lockItem.setLock_object(db.getFullName());
+
+        Thread owner = lock.getOwner();
+        List<Long> sharedLockThreadIds = lock.getSharedLockThreadIds();
+
+        if (owner != null) {
+            lockItem.setLock_mode("EXCLUSIVE");
+            lockItem.setGranted(true);
+            JsonObject ownerInfo = new JsonObject();
+            ownerInfo.addProperty("threadId", owner.getId());
+            ownerInfo.addProperty("threadName", owner.getName());
+            lockItem.setThread_info(ownerInfo.toString());
+
+            // wait start
+            long lockStartTime = lock.getExclusiveLockTime();
+            lockItem.setLock_start_time(lockStartTime);
+        } else if (CollectionUtils.isNotEmpty(sharedLockThreadIds)) {
+            lockItem.setLock_mode("SHARED");
+            lockItem.setGranted(true);
+
+            // lock start
+            long lockStart = ListUtils.emptyIfNull(sharedLockThreadIds).stream().map(lock::getSharedLockTime)
```

Comment:
lock::getSharedLockTime will return -1, if the lock has been released.

Reply:
fixed